src/model/user.rs

```diff
@@ -97,8 +98,18 @@ impl User {
             format!(cdn!("/avatars/{}/{}.jpg"), self.id, av))
     }
 
+    /// Gets user as `Member` of a server.
+    #[cfg(feature="methods")]
+    #[cfg(feature="cache")]
+    pub fn member<G>(&self, guild_id: G) -> Option<Member>
+        where G: Into<GuildId> {
+        let cache = CACHE.read().unwrap();
+
+        cache.get_member(guild_id.into(), self.id).map(|x| { x.clone() })
```

Comment:
This can be `CACHE.read().unwrap().get_member(guild_id.into(), self.id).cloned()`; `cloned` is shorthand for `.map(|x| x.clone())`